srml/transaction-payment/rpc/src/lib.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RPC interface for the t
+//! ransa
+//!ction payment module.
+use std::sync::Arc;
+use serde::{Serialize, Deserialize};
+use codec::Codec;
+use client::blockchain::HeaderBackend;
+use jsonrpc_core::{Error, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use sr_primitives::{
+	generic::BlockId,
+	traits::{Block as BlockT, ProvideRuntimeApi},
+};
+use srml_transaction_payment_rpc_runtime_api::{
+	TransactionPaymentApi as TransactionPaymentRuntimeApi, RuntimeDispatchInfo,
+};
+use self::gen_client::Client as TransactionPaymentClient;
+
+
+/// A wrapper around the request information to query a dispatchable info.
+#[derive(Serialize, Deserialize)]
+pub struct DispatchInfoRequest<AccountId, Extra, Call> {
+	/// Sender aka. origin of the dispatch.
+	pub origin: AccountId,
```

Comment:
For an extrinsic, this is actually `Address`, not `AccountId` (i.e. it can be sent from an `AccountIndex` as well - not 100% sure where the reverse lookup into `AccountId` is done when it hits the call)

Reply:
right -- for the same reason they can have various length actually.

Reply:
Shouldn't this be coming from `Extrinsic` trait or at least be somehow bound to that trait?

Reply:
I think it would be best to accept raw bytes here but it seems you've noticed that already. Then we can just decode to expected type or even pass it raw to the runtime (I prefer the former though, as you can return more meaningful error messages).

We could probably have a wrapper type that does this encoding automatically, i.e. implements `serde::{Ser, De}` for every type that is `scale::{Encode, Decode}`.

Reply:
Totally changed. now we accept the unchecked_extrinsic, totally same as a normal one.